### PR TITLE
Container module knack conversion

### PR DESCRIPTION
**Pull request by derekbekoe**
@yolo3301  Please review. You can use the prompt.ws service below to do some ad-hoc testing of the commands (It takes 2/3 minutes to load so refresh the page afterwards).

Include this fix https://github.com/Azure/azure-cli/pull/5040/files

**Re-recording tests:**
- What's the best way to re-record the test that I'm currently skipping?
Looks like some storage resources are required first. It would be good to add those commands in so the re-recording can be automated.
- Also, for the acr test, can that be automated as well?

### CODE REVIEW

**Comment by tjprescott** [comment]
Related #4929.

File: src/command_modules/azure-cli-container/azure/cli/command_modules/container/_params.py [hunk 1]
```diff
@@ -28,31 +27,30 @@ def environment_variables_format(value):
...
+        c.argument('cpu', type=int, help='The required number of CPU cores of the containers')
+        c.argument('memory', type=float, help='The required memory of the containers in GB')
+        c.argument('os_type', arg_type=get_enum_type(OperatingSystemTypes), help='The OS type of the containers')
+        c.argument('ip_address', arg_type=get_enum_type(IP_ADDRESS_TYPES), help='The IP address type of the container group')
+        c.argument('ports', type=int, nargs='+', default=[80], help='The ports to open')
+        c.argument('restart_policy', arg_type=get_enum_type(ContainerGroupRestartPolicy), help='Restart policy for all containers within the container group')
+        c.argument('command_line', help='The command line to run when the container is started, e.g. \'/bin/bash -c myscript.sh\'')
+        c.argument('environment_variables', nargs='+', options_list=['--environment-variables', '-e'], type=_environment_variables_type, help='A list of environment variable for the container. Space separated values in \'key=value\' format.')
+        c.argument('registry_login_server', arg_group='Image Registry', help='The container image registry login server')
```
Comment (tjprescott) [inline src/command_modules/azure-cli-container/azure/cli/command_modules/container/_params.py]: I would suggest breaking this big block up like so:
```Python
with self.argument_context('container create') as c:
   c.argument(...)
   c.argument(...)

with self.argument_context('container create', arg_group='Image Registry') as c:
   c.argument(...)
   c.argument(...)

with self.argument_context('container create', arg_group='Azure File Volume') as c:
   c.argument(...)
   c.argument(...)
```

In that way it keeps the arguments in the "arg_group" grouped visually together and avoids repeatedly copying the arg_group to each command.

Comment (derekbekoe) [reply src/command_modules/azure-cli-container/azure/cli/command_modules/container/_params.py]: done

**Comment by derekbekoe** [comment]
@tjprescott please take another look

**Comment by tjprescott** [comment]
LGTM!

**Pull request merged**